src/main/scala/com/wavesplatform/state2/diffs/ExecuteContractFunctionTransactionDiff.scala

```diff
@@ -0,0 +1,80 @@
+package com.wavesplatform.state2.diffs
+
+import com.wavesplatform.state2.{Diff, LeaseInfo, Portfolio}
+import com.wavesplatform.state2.reader.StateReader
+import scorex.account.PublicKeyAccount
+import scorex.transaction.ValidationError
+import scorex.transaction.ValidationError._
+import vsys.contract.ExecutionContext
+import vsys.state.opcdiffs.OpcFuncDiffer
+import vsys.transaction.TransactionStatus
+import vsys.transaction.contract.ExecuteContractFunctionTransaction
+import vsys.transaction.proof.{EllipticCurve25519Proof, Proofs}
+
+
+object ExecuteContractFunctionTransactionDiff {
+  def apply(s: StateReader, height: Int)(tx: ExecuteContractFunctionTransaction): Either[ValidationError, Diff] = {
+    if (tx.proofs.proofs.length > Proofs.MaxProofs) {
+      Left(GenericError(s"Too many proofs, max ${Proofs.MaxProofs} proofs"))
+    } else {
+      val sender = EllipticCurve25519Proof.fromBytes(tx.proofs.proofs.head.bytes.arr).toOption.get.publicKey
```

Comment:
```tx.proofs.proofs.head``` is already an instance of Proof. Conversion from bytes may be not efficient.
Use ```asInstanceOf[EllipticCurve25519Proof]``` after verifying the proof type instead.

Reply:
Other diffs use fromBytes. No need to keep consistence here?

Reply:
We may refactor all of them in next patch

Reply:
Sounds good